src/sprightly.ts

```diff
@@ -11,33 +11,41 @@ interface Options {
   cache?: boolean
 }
 
-function parse(file: string, entryPoint: string, data: Data, options: Options) {
-  const parsedFile = file.replaceAll(
+class SprightlyError extends Error {
+  constructor(message: string) {
+    super(message)
+  }
+}
+
+function parse(
+  fileContent: string,
+  filePath: string,
+  data: Data,
+  options: Options,
+) {
+  const parsedFile = fileContent.replaceAll(
     /\{\{(>?)(.*?)\}\}/g,
     (_, isComponent: string, reference: string) => {
       reference = reference.trim()
       if (isComponent) {
-        // if last segement of path is a file
         // we go back one step to get the directory of the file
-        if (path.extname(entryPoint)) {
-          entryPoint = path.normalize(path.join(entryPoint, ".."))
-        }
+        const fileDirectory = path.normalize(path.join(filePath, ".."))
```

Comment:
No need to normalize and you can assume that the path is valid. Also, you can use `path.dirname()` directly. Also, `fileDir` is shorter. Or it might be even enough to use it directly inline without defining a variable.

Reply:
Also, when using the `dirname` util you will no longer need the comment 😄.